fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -226,6 +237,99 @@ def clean_all(self):
         logging.debug("FaProfSession::clean_all()")
         # Delete all log file artifacts
 
+    @staticmethod
+    def validSessionArgs(dictProfTgt):
+        """Determine Profiler session argument status. Return bool"""
+        return ProfSessionArgsStatus.OK in FaProfSession.validateArgs(dictProfTgt)
+
+    @staticmethod
+    def throwOnInvalidSessionArgs(dictProfTgt):
+        """Throw exception on first invalid Profiler session argument."""
+        dictInvalidEnums = FaProfSession.validateArgs(dictProfTgt)
+        if ProfSessionArgsStatus.OK not in dictInvalidEnums:
+            error_enum = next(iter(dictInvalidEnums))
+            error_msg = dictInvalidEnums[error_enum]
+            raise ProfSessionException(error_msg, error_enum)
+
+    @staticmethod
+    def validateArgs(dictProfTgt):
+        """
+        Validates the Profiler Session object's user, target, pwd parameters.
+        Returns a dictionary mapping enums to error msgs.
+        """
+        dictReturn = {}
+        logging.debug(f"validateProfArgs({dictProfTgt}")
+
+        exec_path = dictProfTgt["executeText"]
+        exec_user = dictProfTgt["userText"]
+        exec_pwd = dictProfTgt["dirText"]
+
+        # exec empty?
+        if not exec_path:
+            dictReturn[ProfSessionArgsStatus.EXEC_EMPTY] = s.PROF_ARG_EXEC_EMPTY
+
+        # exist?
+        elif not os.path.exists(exec_path):
+            dictReturn[ProfSessionArgsStatus.EXEC_DOESNT_EXIST] = (
+                exec_path + s.PROF_ARG_EXEC_DOESNT_EXIST
+            )
+
+        # executable?
+        elif not os.access(exec_path, os.X_OK):
+            dictReturn[ProfSessionArgsStatus.EXEC_NOT_EXECUTABLE] = (
+                exec_path + s.PROF_ARG_EXEC_NOT_EXECUTABLE
+            )
+
+        if ProfSessionArgsStatus.OK in dictReturn:
+            logging.debug("FaProfSession::validateArgs() --> exec verified")
+
+        # user?
+        try:
+            if exec_user:
+                pwd.getpwnam(exec_user)
+        except KeyError as e:
+            logging.debug(f"User {exec_user} does not exist: {e}")
+            dictReturn[ProfSessionArgsStatus.USER_DOESNT_EXIST] = (
+                exec_user + s.PROF_ARG_USER_DOESNT_EXIST
+            )
+
+        if ProfSessionArgsStatus.OK in dictReturn:
+            logging.debug("FaProfSession::validateArgs() --> user verified")
```

Comment:
Same as above, this debug statement will never be hit. I'd suggest removing it and just debug logging the results at the end.

Reply:
Agreed. Removed.